src/client.h

```diff
@@ -114,23 +114,43 @@ struct MeshUpdateResult
 	}
 };
 
-class MeshUpdateThread : public UpdateThread
+class MeshUpdateThread : public PoolableUpdateThread<u8>
 {
-private:
-	MeshUpdateQueue m_queue_in;
+public:
+	typedef	u8 MessageType;
 
 protected:
-	virtual void doUpdate();
+	virtual void doUpdate(const MessageType &);
 
 public:
+	MeshUpdateThread(const std::string &name, Semaphore &sem,
+		MutexedQueue< ThreadPoolMessage<MessageType> > &msgqueue,
+		MeshUpdateQueue &queue_in,
+		MutexedQueue<MeshUpdateResult> &queue_out, v3s16 &camera_offset) :
```

Comment:
Lots of non const references here too. Either make them const (if possible), or make them pointers.